engine/src/main/java/org/hibernate/search/engine/logging/impl/Log.java

```diff
@@ -275,4 +274,10 @@ SearchException unableToParseTemporal(@FormatWith(SimpleNameClassFormatter.class
 
 	@Message(id = ID_OFFSET_2 + 58, value = "Invalid %1$s value: expected either a Number or a String that can be parsed into a %1$s. %2$s")
 	SearchException invalidNumberPropertyValue(@FormatWith(SimpleNameClassFormatter.class) Class<? extends Number> type, String nestedErrorMessage, @Cause Exception cause);
+
+	@Message(id = ID_OFFSET_2 + 59, value = "Invalid '%1$s' value for type '%2$s'.")
+	SearchException invalidStringForType(String value, @FormatWith(SimpleNameClassFormatter.class) Class<?> type);
```

Comment:
Maybe pass the cause in an additional `@Cause Exception cause` parameter?
IIRC you can pass null and it will just work (for your Character validation).

Reply:
Also you could use the same method for both the enum and other types, it's not a big deal if we don't use the simple name :)

Reply:
OK to add the cause. On the other hand I would like to keep a different message for Enum parsing error.